model/ftn/ww3_prtide.ftn

```diff
@@ -501,14 +505,15 @@
                 IF (ABS(TIDAL_CONST(IX,IY,INDMAX(I),1,1)) .GT.MAXVALCON(I) .OR. &
                     ABS(TIDAL_CONST(IX,IY,INDMAX(I),2,1)) .GT.MAXVALCON(I)) THEN
                   TIDEOK = 0
-                  WRITE(NDSO,*) 'BAD: ', INDMAX(I), MAXVALCON(I), '/', ABS(TIDAL_CONST(IX,IY,INDMAX(I),1:2,1))
+                  WRITE(NDSO,'(A,I8,F10.2,A,F10.2)') 'BAD: ', INDMAX(I), MAXVALCON(I), &
+                        '/', ABS(TIDAL_CONST(IX,IY,INDMAX(I),1:2,1))
                 END IF
                 BADPOINTS(IX,IY) = BADPOINTS(IX,IY) + (1-TIDEOK)
               END DO
 
               IF (BADPOINTS(IX,IY).GT.0) THEN 
                 NBAD = NBAD +1
-                WRITE(NDSE,*) 'BAD POINT:',IX,IY,NBAD, &
+                WRITE(NDSE,'(A,I8,I8,F10.2,A,F10.2)') 'BAD POINT:',IX,IY,NBAD, &
```

Comment:
The format specifier is wrong here; I think you need another I8 (for NBAD), i.e.
``` '(A,I8,I8,I8,F10.2,A,F10.2)') ```

Reply:
Furthermore, the slice `TIDAL_CONST(IX,IY,:,1,1)` is multiple values, not a single float, and this is causing runtime errors as there are not enough format specifiers. An example line (produced by using list directed I/O) looks like this:

```
BAD POINT:           2           1           1  -327.669983       2.33866073E-12   2.27044360E-12   2.26722764E-12   2.25777578E-12   2.49276901E-12   2.32712092E-12   2.32340861E-12   2.39584112E-12   2.06419994E-12   2.35532514E-12   2.40584549E-12   2.05155250E-12   2.04318875E-12   2.16234626E-12   2.15818834E-12   2.20166940E-12   1.93449011E-12   2.05661768E-12   2.03594628E-12 ##  -327.669983       2.33866073E-12   2.27044360E-12   2.26722764E-12   2.25777578E-12   2.49276901E-12   2.32712092E-12   2.32340861E-12   2.39584112E-12   2.06419994E-12   2.35532514E-12   2.40584549E-12   2.05155250E-12   2.04318875E-12   2.16234626E-12   2.15818834E-12   2.20166940E-12   1.93449011E-12   2.05661768E-12   2.03594628E-12
BAD:        1      1.00/    327.67```

Reply:
I've pushed the corrected version but I don't have any BAD POINT statements in the log files.. so I'm not able to confirm it's ok now. Do you have this output with regtest ww3_tp2.18 ?

Reply:
@mickaelaccensi I get lots of "BAD POINT" output when I run that test. It's odd that you don't get any.
Also, the order of the "BAD POINT" output is still not reproducible between two identical runs.

I have attached a zip file here with the output from two runs of ww3_tp2.18/TIDE_MPI: [ww3_prtide_current_out.zip](https://github.com/NOAA-EMC/WW3/files/6584486/ww3_prtide_current_out.zip)

Reply:
I notice that there are a lot of values of 327.67 or -327.67 in those "BAD" points that I sent you - they looks suspiciously like the netCDF `fille_value` of -32678 (2^15) scaled by 0.01... Are there perhaps some invalid points being read from the netCDF?

Reply:
I don't get any "BAD POINT" in the ww3_tp2.18 ww3_prtide_current.out either.  I did run the two tests twice and confirmed they reproduce each other now for NCEP.